compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -166,14 +166,9 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
         InstrumentedParts.notCovered(tree)
 
     private def tryInstrument(tree: Select)(using Context): InstrumentedParts =
-      val sym = tree.symbol
       val transformed = cpy.Select(tree)(transform(tree.qualifier), tree.name)
-      if canInstrumentParameterless(sym) then
```

Comment:
thanks these comments make sense, and indeed as I do more testing I see it's not this simple :) 

hopefully will report back after tackling this further.

Reply:
@TheElectronWill so I'm able to reproduce the issue neatly locally now, and then use my local dotty build in sbt-scoverage-samples in order to clearly see the coverage issue and how my local changes impact it
<img width="605" alt="Screen Shot 2023-04-18 at 7 58 53 AM" src="https://user-images.githubusercontent.com/556288/232818009-2a45d83c-127d-4d96-aff9-2aecd3915a99.png">


I am able to get the right side of the flatMap/for comprehension to show up in the measurements file but not arbitrarily deep (so the right hand side of `a <- List(1)` will be green but the right hand side of the next line `b <- List(2)` won't be. 

& also it somewhat seems fundamentally wrong to me, because we are getting a statement for the whole Apply flatMap tree around everything - 

```
 for {\n s <- Future.successful("foo")\n t <- Future.successful("barnum")\n } yield {\n "yield me"\n } 
```

and that is properly getting added to the measurements file when it is called.

so I actually think the "correct" solution would be to stop these other statements (that are inside the flatMap/for comprehension statement) from showing up entirely.  does that make sense to you?

Reply:
> so I actually think the "correct" solution would be to stop these other statements (that are inside the flatMap/for comprehension statement) from showing up entirely. does that make sense to you?

But then, how would the coverage check that all of the for-comprehension is executed? The first flatMap could fail, and in that case the rest of the for-comprehension must not be 100% covered.